packages/server/src/simulation.ts

```diff
@@ -43,6 +43,30 @@ export function simulation(definitions: Record<string, Simulator>): Effect<Simul
         services.push({ name, url: `${protocol}://localhost:${address.port}` });
       }
 
+      let store = slice.slice("store");
+      let { scenarios } = behaviors;
+
+      scope.spawn(map(slice.slice("scenarios"), slice => function*() {
+        try {
+          let { name } = slice.get();
+          let fn = scenarios[name];
+          assert(fn, `unknown scenario ${name}`);
+
+          let data = yield fn(store);
+          slice.update(state => ({
```

Comment:
definitely nice to have some abstraction over this `update` error.  Something like an 

`type Either<E, A> = Left<E> | Right<A>`

With the 2 alternative paths.

But this is totally fine for now